Source/WebCore/platform/graphics/FlatDynamicRangeLimit.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2025 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <wtf/ArgumentCoder.h>
+
+namespace WTF {
+class TextStream;
+}
+
+namespace WebCore {
+
+namespace Style {
+struct DynamicRangeLimit;
+}
+
+enum class ApplicableDynamicRangeLimit : uint8_t {
+    Standard,
+    ConstrainedHigh,
+    NoLimit
+};
+
+class FlatDynamicRangeLimit {
+public:
+    constexpr FlatDynamicRangeLimit() = default;
+
+    static constexpr FlatDynamicRangeLimit standard() { return FlatDynamicRangeLimit(standardValue); }
+    static constexpr FlatDynamicRangeLimit constrainedHigh() { return FlatDynamicRangeLimit(constrainedHighValue); }
+    static constexpr FlatDynamicRangeLimit noLimit() { return FlatDynamicRangeLimit(noLimitValue); }
+
+    // [standard, constrainedHigh] mapped to [0, 0.5],
+    // [constrainedHigh, noLimit] mapped to [0.5, 1].
+    constexpr float value() const { return m_value; }
+
+    constexpr ApplicableDynamicRangeLimit applicableValue() const
+    {
+        if (value() < standardAndConstrainedHighThreshold)
+            return ApplicableDynamicRangeLimit::Standard;
+        if (value() < constrainedHighAndNoLimitThreshold)
+            return ApplicableDynamicRangeLimit::ConstrainedHigh;
+        return ApplicableDynamicRangeLimit::NoLimit;
+    }
+
+    constexpr auto operator<=>(const FlatDynamicRangeLimit&) const = default;
+
+private:
+    friend struct IPC::ArgumentCoder<WebCore::FlatDynamicRangeLimit, void>;
+    friend Style::DynamicRangeLimit;
+
+    constexpr FlatDynamicRangeLimit(float value) : m_value(value) { }
```

Comment:
Why is this constrictor marked as `private`? I think we do not like to use `friend` much unless it is really necessary. Is there a reason for not making `public`?